### PR TITLE
[netcore] Do not inline ThrowHelpers, fix SkipFrames handling.

### PR DESCRIPTION
**Pull request by EgorBo**
1) Do not inline ThrowHelpers (e.g. `static void ThrowException() => throw new Exception();`)
2) `get_trace` incorrectly handles `skip` argument
3) fix broken ci (`dl-test-assets.py` is not python2 compatible any more)

Fixes https://github.com/mono/mono/issues/15182
Fixes https://github.com/mono/mono/issues/15188

### CODE REVIEW

File: netcore/Makefile [hunk 1]
```diff
@@ -126,7 +126,7 @@ update-tests-corefx: corefx/.stamp-dl-corefx-tests-$(NETCORETESTS_VERSION)
 
 corefx/.stamp-dl-corefx-tests-$(NETCORETESTS_VERSION):
 	curl $(TEST_ASSETS_URL) --output corefx-test-assets.xml
-	python dl-test-assets.py corefx-test-assets.xml $(FEED_BASE_URL) corefx/tests
```
Comment (vargaz) [inline netcore/Makefile]: Unrelated changes ?

File: mono/mini/mini-exceptions.c [hunk 2]
```diff
@@ -1141,7 +1141,7 @@ ves_icall_get_trace (MonoException *exc, gint32 skip, MonoBoolean need_file_info
 		}
 
 		mono_debug_free_source_location (location);
-		mono_array_setref_internal (res, i, sf);
+		mono_array_setref_internal (res, i - skip, sf);
```
Comment (vargaz) [inline mono/mini/mini-exceptions.c]: Why did this work previously ?

Comment (EgorBo) [reply mono/mini/mini-exceptions.c]: I guess it never worked if `skip` was > 0, e.g. this crashes on regular mono:
```csharp
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;

namespace HelloWorld
{
    class Program
    {
        private static Exception InvokeException()
        {
            try
            {
                ThrowException();
                return null;
            }
            catch (Exception ex)
            {
                return ex;
            }
        }
        
        [MethodImpl(MethodImplOptions.NoInlining)]
        private static void ThrowException() => throw new Exception();
        
        static void Main(string[] args)
        {
            int skipFrames = 1;
            Exception ex = InvokeException();
            var exceptionStackTrace = new StackTrace(ex, skipFrames);
            Console.WriteLine(exceptionStackTrace);
        }
    }
}
```

Comment (filipnavara) [reply mono/mini/mini-exceptions.c]: Presumably because in most cases it was called with zero. In non-netcore build all the default constructors pass `METHODS_TO_SKIP == 0`.

File: netcore/Makefile (hunk 1 above)
Comment (EgorBo) [reply netcore/Makefile]: @vargaz on our CI `python` stands for Python 2 (default for macOS) and https://github.com/mono/mono/pull/15524 made that script python3 only

Comment (vargaz) [reply netcore/Makefile]: I mean the change is unrelated to the PR.

Comment (EgorBo) [reply netcore/Makefile]: I understand but without it I will not be able to see CI results for this PR 🙂 all netcore lanes are currently crash on the `download corefx tests` step in master. I can file a separate PR for it.

**Comment by EgorBo** [comment]
/azp run
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by steveisok** [comment]
@monojenkins build failed
